src/porepy/numerics/ad/equation_system.py

```diff
@@ -477,6 +477,42 @@ def create_variables(
 
         return merged_variable
 
+    def remove_variables(self, variables: VariableList) -> None:
```

Comment:
The method will change the counting of the dofs, but not update the variable state. Hence it is not safe to use it after the variables have been initialized, not even with zero values (the vector size will be unexpectedly large). I think a clarification of the intended usage is in place, together with either clear warnings when to use the method or extended functionality (and tests).

Reply:
Which vector size are you referring to?

Reply:
Variable state vectors, so basically my comment is the same as above.